TLM/TLM/UI/Translation.cs

```diff
@@ -12,77 +17,254 @@ namespace TrafficManager.UI {
     using Util;
 
     public class Translation {
-        public static readonly IDictionary<string, string> LanguageLabels;
-        public static readonly IList<string> AvailableLanguageCodes;
+        /// <summary>
+        /// Defines the order of language selector dropdown.
+        /// </summary>
+        public static readonly IList<string> AvailableLanguageCodes
+            = new List<string> {
+                                   "en",
+                                   "en-gb",
+                                   "de",
+                                   "es",
+                                   "fr",
+                                   "hu",
+                                   "it",
+                                   "ja",
+                                   "ko",
+                                   "nl",
+                                   "pl",
+                                   "pt",
+                                   "ru",
+                                   "zh", // Mainland China (simplified)
+                                   "zh-tw", // Taiwanese (traditional)
+                               };
+
+        /// <summary>
+        /// Defines column order in CSV, but not the dropdown order.
+        /// Mapping from translation languages (first row of CSV export) to language locales used in
+        /// <see cref="AvailableLanguageCodes"/> above.
+        /// </summary>
+        private static readonly Dictionary<string, string> csv_columns_to_locales_
+            = new Dictionary<string, string>() {
+                                                   { "English", DEFAULT_LANGUAGE_CODE },
```

Comment:
Replace `DEFAULT_LANGUAGE_CODE` with `en`?

I can see why use it, but it seems a bit strange in this context. I would expect the constant to be used to choose the associated language _from_ the list, rather than being in the list itself.

Also, assuming `en` is going to be Americanised, should its caption become (here and in csv) "English (American)"?

Reply:
We could just put a `// DEFAULT_LANGUAGE_CODE` after that entry.

Also we should maybe arrange it so that both English languages are next to each other.  As the current language is selected by default it shouldn't matter if English isn't top of the list?

Reply:
We can assume "en" is "en-US" or rename it properly, whatever you prefer. Renaming is a tiny bit more work though.
Language titles are now a string in the translations "I18n Language Name", i might rename it later again, when we figure out some consistent string naming scheme.